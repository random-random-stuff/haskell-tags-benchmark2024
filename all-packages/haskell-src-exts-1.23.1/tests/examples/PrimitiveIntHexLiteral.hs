{-# LANGUAGE MagicHash #-}
minInt = I# (0x8000000000000000#)
