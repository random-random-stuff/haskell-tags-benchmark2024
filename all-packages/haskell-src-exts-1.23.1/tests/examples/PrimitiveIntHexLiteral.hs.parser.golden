ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan =
            SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 1 1 3 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 1 1 1 1
            , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 1 2 1
            , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 1 2 1
            , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 1 2 1
            , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 3 1 3 1
            , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 3 1 3 1
            ]
        }
      Nothing
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 1 1 1 27
            , srcInfoPoints =
                [ SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 1 1 1 13
                , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 1 24 1 27
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 1 14 1 23
                , srcInfoPoints = []
                }
              "MagicHash"
          ]
      ]
      []
      [ PatBind
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 1 2 34
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 1 2 7
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 1 2 7
                  , srcInfoPoints = []
                  }
                "minInt"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 8 2 34
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 8 2 9 ]
               }
             (App
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 10 2 34
                  , srcInfoPoints = []
                  }
                (Con
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 10 2 12
                     , srcInfoPoints = []
                     }
                   (UnQual
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 10 2 12
                        , srcInfoPoints = []
                        }
                      (Ident
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 10 2 12
                           , srcInfoPoints = []
                           }
                         "I#")))
                (Paren
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 13 2 34
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 13 2 14
                         , SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 33 2 34
                         ]
                     }
                   (Lit
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 14 2 33
                        , srcInfoPoints = []
                        }
                      (PrimInt
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/PrimitiveIntHexLiteral.hs" 2 14 2 33
                           , srcInfoPoints = []
                           }
                         9223372036854775808
                         "0x8000000000000000")))))
          Nothing
      ]
  , []
  )
